src/messages.rs

```diff
@@ -1,17 +1,38 @@
-use ft_io::*;
+use codec::Encode;
+use ft_main_io::*;
 use gstd::{msg, ActorId};
 
-pub async fn transfer_tokens(token_id: &ActorId, from: &ActorId, to: &ActorId, amount: u128) {
-    let _transfer_response = msg::send_for_reply(
-        *token_id,
-        FTAction::Transfer {
-            from: *from,
-            to: *to,
-            amount,
+/// Transfers `amount` tokens from `sender` account to `recipient` account.
+/// Arguments:
+/// * `transaction_id`: associated transaction id
+/// * `from`: sender account
+/// * `to`: recipient account
+/// * `amount`: amount of tokens
+pub async fn transfer_tokens(
+    transaction_id: u64,
+    token_address: &ActorId,
+    from: &ActorId,
+    to: &ActorId,
+    amount_tokens: u128,
+) -> Result<(), ()> {
+    let reply = msg::send_for_reply_as::<_, FTokenEvent>(
```

Comment:
I still prefer avoiding turbo-fish 😄 
```suggestion
    let reply: FTokenEvent = msg::send_for_reply_as(
```